File: frontend/bindings/cleanC/services/system/systemservice.js
// @ts-check
// 此文件由 wails3 generate bindings 自动生成
// 开发时请运行 wails3 dev 或 wails3 generate bindings 来更新

import {Call as $Call} from "/wails/runtime.js";

/**
 * @returns {Promise<{hostname: string, os: string, arch: string, numCPU: number}>}
 */
export function GetSystemInfo() {
    return $Call.ByName("cleanC/services/system.SystemService.GetSystemInfo");
}

/**
 * @returns {Promise<string>}
 */
export function GetHostname() {
    return $Call.ByName("cleanC/services/system.SystemService.GetHostname");
}

/**
 * @returns {Promise<string>}
 */
export function GetOS() {
    return $Call.ByName("cleanC/services/system.SystemService.GetOS");
}

/**
 * @returns {Promise<string>}
 */
export function GetArch() {
    return $Call.ByName("cleanC/services/system.SystemService.GetArch");
}

/**
 * @returns {Promise<number>}
 */
export function GetNumCPU() {
    return $Call.ByName("cleanC/services/system.SystemService.GetNumCPU");
}

File: internal/service/systemservice.go
package service

import (
	"os"
	"runtime"
)

type SystemService struct{}

func NewSystemService() *SystemService {
	return &SystemService{}
}

func (s *SystemService) GetHostname() (string, error) {
	return os.Hostname()
}

func (s *SystemService) GetOS() string {
	return runtime.GOOS
}

func (s *SystemService) GetArch() string {
	return runtime.GOARCH
}

func (s *SystemService) GetNumCPU() int {
	return runtime.NumCPU()
}

type SystemInfo struct {
	Hostname string `json:"hostname"`
	OS       string `json:"os"`
	Arch     string `json:"arch"`
	NumCPU   int    `json:"numCPU"`
}

func (s *SystemService) GetSystemInfo() (*SystemInfo, error) {
	hostname, err := os.Hostname()
	if err != nil {
		return nil, err
	}
	return &SystemInfo{
		Hostname: hostname,
		OS:       runtime.GOOS,
		Arch:     runtime.GOARCH,
		NumCPU:   runtime.NumCPU(),
	}, nil
}
